### PR TITLE
24141 - Added founding date not changed from Colin

### PR DESCRIPTION
**Pull request by ArwenQin**
*Issue #:* /bcgov/entity#24141

*Description of changes:*
- Retain the founding date from Colin for the continuation filing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-755-z1jrh228.web.app

**Comment by ArwenQin** [comment]
try:
https://business-create-dev--pr-755-z1jrh228.web.app/continuation-in-authorization?id=TccWOxq2Ov&accountid=3040

now the foundingDate is in the same format as Colin:
image

File: src/components/ContinuationIn/ExtraproRegistration.vue [hunk 1]
```diff
@@ -409,7 +409,8 @@ export default class ExtraproRegistration extends Mixins(DateMixin) {
     }
 
     this.business = {
-      bcRegistrationDate: this.dateToYyyyMmDd(this.apiToDate(businessInfo.foundingDate)),
+      bcRegistrationDate: businessInfo.foundingDate,
+      bcRegistrationDateFormatted: this.dateToYyyyMmDd(this.apiToDate(businessInfo.foundingDate)),
```
Comment (severinbeauvais) [inline src/components/ContinuationIn/ExtraproRegistration.vue:413]: I wouldn't add this to the business object.

How about creating a getter (computed) to return this, that you can use in the template and rules above?

File: src/interfaces/store-interfaces/state-interfaces/existing-business-info-interface.ts [hunk 2]
```diff
@@ -4,7 +4,8 @@ export interface ExistingBusinessInfoIF {
   affidavitFile?: File // only used by UI
   affidavitFileKey?: string
   affidavitFileName?: string
-  bcRegistrationDate?: string // expro only (YYYY-MM-DD)
+  bcRegistrationDate?: string
```
Comment (severinbeauvais) [inline src/interfaces/store-interfaces/state-interfaces/existing-business-info-interface.ts:7]: Please restore the comment but update it to what it is (ISO date-time).

File: src/mixins/filing-template-mixin.ts [hunk 3]
```diff
@@ -419,9 +419,9 @@ export default class FilingTemplateMixin extends Mixins(AmalgamationMixin, DateM
 
     // Add expro business information.
     if (this.getExistingBusinessInfo?.mode === 'EXPRO') {
-      const foundingDate = this.yyyyMmDdToDate(this.getExistingBusinessInfo?.bcRegistrationDate)
+      const foundingDate = this.getExistingBusinessInfo?.bcRegistrationDate
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:422]: How about moving the right hand side into line 424 (same as lines 425-426)?

File: src/mixins/filing-template-mixin.ts [hunk 4]
```diff
@@ -464,14 +464,15 @@ export default class FilingTemplateMixin extends Mixins(AmalgamationMixin, DateM
 
     // restore existing business information
     if (continuationIn.foreignJurisdiction) {
-      const foundingDate = this.apiToDate(continuationIn.business?.foundingDate)
+      const foundingDate = continuationIn.business?.foundingDate
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:467]: Same as above.

File: src/components/ContinuationIn/ExtraproRegistration.vue (hunk 1 above)
Comment (ArwenQin) [reply src/components/ContinuationIn/ExtraproRegistration.vue]: Updated this, thanks.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-755-z1jrh228.web.app

**Review by severinbeauvais — APPROVED** [review]
Nice and clean. I like it!

Please update the app version and then go ahead and merge it.

**Comment by JazzarKarim** [comment]
Probably not intentional, but I think here the version should've been 5.13.1 as this is a small patch. Please watch out for next time, thank you!
1_S0S5Qv9AK3mc1M3Vq39nPQ

**Comment by severinbeauvais** [comment]
Yes, this was an oopsie. I said it was ok to leave it.

________________________________

**Comment by ArwenQin** [comment]
Thank you! I will pay attention next time

**Pull request merged**